### PR TITLE
Bazel -> CMake cleanup

### PR DESCRIPTION
**Pull request by GMNGeoffrey**
- Make a bunch of member functions free functions.
- Delete a bunch of pretty useless helpers and refactor for increased
  reuse.
- Be consistent about function argument defaults.
- Be more explicit about quoting things.

### CODE REVIEW

File: build_tools/bazel_to_cmake/bazel_to_cmake_converter.py [hunk 1]
```diff
@@ -27,207 +27,151 @@
...
-    return self._convert_option_block("FLATTEN", flatten)
-
-  def _convert_file_list_block(self, list_name, files):
-    #  list_name
-    #    "file_1.h"
-    #    "file_2.h"
-    #    "file_3.h"
-    if not files:
-      return ""
-    files_list = "\n".join([f'    "{file}"' for file in sorted(files)])
```
Comment (ScottTodd) [inline build_tools/bazel_to_cmake/bazel_to_cmake_converter.py:115]: IIRC, the quoting for file names was first introduced in the hand-ported CMake files and then I copied that style over to the scripts. I'm not sure if there's value in continuing that styling.

If you proceed with removing the quotes around file names, can you also update the documentation to match?

https://github.com/google/iree/blob/cf1047322db8e591c25cc27de954d8d68173d9e4/build_tools/cmake/iree_cc_library.cmake#L42-L66

https://github.com/google/iree/blob/cf1047322db8e591c25cc27de954d8d68173d9e4/build_tools/cmake/iree_cc_binary.cmake#L35-L55

https://github.com/google/iree/blob/cf1047322db8e591c25cc27de954d8d68173d9e4/build_tools/cmake/external_cc_library.cmake#L46-L78

etc.

---

OR you could keep the style changes out of this commit/PR and make them in isolation.

**Review by ScottTodd — CHANGES_REQUESTED** [review]

File: build_tools/bazel_to_cmake/bazel_to_cmake_converter.py (hunk 1 above)
Comment (GMNGeoffrey) [reply build_tools/bazel_to_cmake/bazel_to_cmake_converter.py:115]: Yeah I didn't intend to do the style change as part of this, but while trying to unify things discovered what appeared to be arbitrary quoting distinctions between different types of arguments. In classic CMake style, searching around reveals that there are no official recommendations on when to quote things and the unofficial recommendations from people are contradictory. So... I just picked something and it seems to all build fine without quotes. If someone with past CMake experience has a strong opinion on when things should be quoted, I'm happy to implement that. I'll save the manual updates until we decide where this is going though. I could also start out with separate quoted/unquoted variations for each string function and preserve the existing style

Comment (GMNGeoffrey) [reply build_tools/bazel_to_cmake/bazel_to_cmake_converter.py:115]: I added optional quoting to the helper functions and restored the previous quoting behavior. We can continue to discuss what the right thing to do here is.

Comment (ScottTodd) [reply build_tools/bazel_to_cmake/bazel_to_cmake_converter.py:115]: SG. I'm not against the quoting changes, but I'd prefer for them to be made in a separate commit that includes the documentation updates.

**Pull request merged**